table/table_test.cc

```diff
@@ -1171,6 +1172,123 @@ class PlainTableTest : public TableTest {};
 class TablePropertyTest : public testing::Test {};
 class BBTTailPrefetchTest : public TableTest {};
 
+// The helper class to test the file checksum
+class FileChecksumHelper {
+ public:
+  FileChecksumHelper(bool convert_to_internal_key = false)
+      : convert_to_internal_key_(convert_to_internal_key) {
+    sink_ = new test::StringSink();
+  }
+  ~FileChecksumHelper() {}
+
+  void CreateWriteableFile() {
+    file_writer_.reset(test::GetWritableFileWriter(sink_, "" /* don't care */));
+  }
+
+  void SetFileChecksumMethod(SstFileChecksum* checksum_cal) {
+    if (file_writer_ != nullptr) {
+      file_writer_->SetFileChecksumMethod(checksum_cal);
+    }
+  }
+
+  WritableFileWriter* GetFileWriter() { return file_writer_.get(); }
+
+  Status ResetTableBuilder(std::unique_ptr<TableBuilder> builder) {
+    assert(builder != nullptr);
+    table_builder_ = std::move(builder);
+    return Status::OK();
+  }
+
+  void AddKVtoKVMap(int num_entries) {
+    Random rnd(test::RandomSeed());
+    for (int i = 0; i < num_entries; i++) {
+      std::string v;
+      test::RandomString(&rnd, 100, &v);
+      kv_map_[test::RandomKey(&rnd, 20)] = v;
+    }
+  }
+
+  Status WriteKVAndFlushTable() {
+    for (const auto kv : kv_map_) {
+      if (convert_to_internal_key_) {
+        ParsedInternalKey ikey(kv.first, kMaxSequenceNumber, kTypeValue);
+        std::string encoded;
+        AppendInternalKey(&encoded, ikey);
+        table_builder_->Add(encoded, kv.second);
+      } else {
+        table_builder_->Add(kv.first, kv.second);
+      }
+      EXPECT_TRUE(table_builder_->status().ok());
+    }
+    Status s = table_builder_->Finish();
+    file_writer_->Flush();
+    EXPECT_TRUE(s.ok());
+
+    EXPECT_EQ(sink_->contents().size(), table_builder_->FileSize());
+    return s;
+  }
+
+  uint32_t GetFileChecksum() { return table_builder_->GetFileChecksum(); }
+
+  const char* GetFileChecksumName() {
+    return table_builder_->GetFileChecksumName();
+  }
+
+  Status CalculateFileChecksum(SstFileChecksum* file_checksum_cal,
+                               uint32_t* checksum) {
+    assert(file_checksum_cal != nullptr);
+    cur_uniq_id_ = checksum_uniq_id_++;
+    test::StringSink* ss_rw =
+        rocksdb::test::GetStringSinkFromLegacyWriter(file_writer_.get());
+    file_reader_.reset(test::GetRandomAccessFileReader(
+        new test::StringSource(ss_rw->contents())));
+    char* scratch = nullptr;
+    scratch = new char[2048];
```

Comment:
Nit: you can use `std::unique_ptr<char[]>` to avoid explicitly deleting.

Reply:
@riversand963 changed accordingly.